test/gtest/uct/test_md.cc

```diff
@@ -1036,6 +1047,63 @@ UCS_TEST_SKIP_COND_P(test_md, rkey_compare,
 
 UCT_MD_INSTANTIATE_TEST_CASE(test_md)
 
+class test_md_odp : public test_md {
+public:
+    void test_odp() {
+        ucs_memory_type_t mem_type = UCS_MEMORY_TYPE_HOST;
+        ucs_status_t status;
+        uct_mem_h memh;
+        void *address;
+
+        if (!(md_attr().reg_nonblock_mem_types & UCS_BIT(mem_type))) {
+            UCS_TEST_SKIP_R(mem_buffer::mem_type_name(mem_type) +
+                            " memory registration is not supported by" +
+                            std::string(GetParam().md_name));
+        }
+
+        for (unsigned i = 1; i <= 300; ++i) {
+            size_t size = UCS_KBYTE * i;
+
+            std::vector<char> fill_buffer(size, 0);
+            ucs::fill_random(fill_buffer);
+
+            alloc_memory(&address, size, &fill_buffer[0], UCS_MEMORY_TYPE_HOST);
+
+            status = reg_mem(UCT_MD_MEM_ACCESS_ALL | UCT_MD_MEM_FLAG_NONBLOCK,
+                             address, size, &memh);
+
+            ASSERT_UCS_OK(status);
+            ASSERT_TRUE(memh != UCT_MEM_HANDLE_NULL);
+            check_memory(address, &fill_buffer[0], size, mem_type);
+
+            status = uct_md_mem_dereg(md(), memh);
+            ASSERT_UCS_OK(status);
+            check_memory(address, &fill_buffer[0], size, mem_type);
+
+            free_memory(address, mem_type);
+        }
+    }
+};
+
+UCS_TEST_SKIP_COND_P(test_md_odp, reg, !check_caps(UCT_MD_FLAG_REG)) {
+    test_odp();
+}
+
+UCT_MD_INSTANTIATE_TEST_CASE(test_md_odp)
+
+class test_md_odp_no_devx : public test_md_odp {
+public:
+    void init() override {
+        modify_config("IB_MLX5_DEVX", "no");
+    }
+};
+
+UCS_TEST_SKIP_COND_P(test_md_odp_no_devx, reg, !check_caps(UCT_MD_FLAG_REG)) {
+    test_odp();
+}
+
+_UCT_MD_INSTANTIATE_TEST_CASE(test_md_odp_no_devx, ib)
```

Comment:
seems to be ib specific, why is it in common `test_md`?

Reply:
I found that ODP is tested already, so test part was significantly changed